src/main/java/com/redhat/syseng/soleng/rhpam/processmigration/service/impl/MigrationServiceImpl.java

```diff
@@ -15,128 +15,215 @@
 import org.jboss.logging.Logger;
 import org.kie.server.api.model.admin.MigrationReportInstance;
 
-import com.redhat.syseng.soleng.rhpam.processmigration.model.Execution.ExecutionStatus;
 import com.redhat.syseng.soleng.rhpam.processmigration.model.Execution.ExecutionType;
-import com.redhat.syseng.soleng.rhpam.processmigration.model.Migration;
+import com.redhat.syseng.soleng.rhpam.processmigration.jpa.model.Migration;
+import com.redhat.syseng.soleng.rhpam.processmigration.jpa.model.MigrationReport;
 import com.redhat.syseng.soleng.rhpam.processmigration.model.MigrationDefinition;
-import com.redhat.syseng.soleng.rhpam.processmigration.model.Plan;
+import com.redhat.syseng.soleng.rhpam.processmigration.jpa.model.Plan;
+import com.redhat.syseng.soleng.rhpam.processmigration.model.Execution;
 import com.redhat.syseng.soleng.rhpam.processmigration.model.ProcessInstanceMigration;
 import com.redhat.syseng.soleng.rhpam.processmigration.model.exceptions.PlanNotFoundException;
 import com.redhat.syseng.soleng.rhpam.processmigration.service.KieService;
 import com.redhat.syseng.soleng.rhpam.processmigration.service.MigrationService;
 import com.redhat.syseng.soleng.rhpam.processmigration.service.PersistenceService;
 import com.redhat.syseng.soleng.rhpam.processmigration.service.PlanService;
+import java.util.Scanner;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+import javax.transaction.Transactional;
 
 @ApplicationScoped
 public class MigrationServiceImpl implements MigrationService {
 
     private static final Logger logger = Logger.getLogger(MigrationServiceImpl.class);
 
+    @PersistenceContext(unitName = "migration-unit")
+    EntityManager em;
+
+    
+
     @Inject
     private PlanService planService;
-    @Inject
-    private PersistenceService<Migration> migrationPersistenceService;
-    @Inject
-    private PersistenceService<ProcessInstanceMigration> piMigrationPersistenceService;
+    //@Inject
+    //private PersistenceService<Migration> migrationPersistenceService;
+    //@Inject
+    //private PersistenceService<ProcessInstanceMigration> piMigrationPersistenceService;
     @Inject
     private KieService kieService;
 
     private static final String RESULTS_QUERY = "";
 
+    //Used by Junit test where em can't be injected
```

Comment:
Find another way. Do not change the implementation only for the tests

Reply:
Let's leave this here for now, otherwise the junit won't work because during testing stage, the code is not in a container, so em won't be injected. 
Maybe later using arquillian runAs can simulate the real container work, but that need more research ( I did some basic testing, didn't work)